spec/requests/post_spec.rb

```diff
@@ -33,6 +33,43 @@
     end
   end
 
+  describe 'GET index' do
+    context 'when user is authenticated' do
+      let!(:user) { create(:user) }
+      let!(:post1) { create(:post) }
+      let!(:post2) { create(:post) }
```

Comment:
Instead of creating just 2 posts, you can use `create_list` method from FactoryBot.
Something like:

```create_list(create_list(:post, 6, user:)```

There, you create 6 posts using the factory and associating it with your already created `user`